src/libs/actions/Policy.js

```diff
@@ -802,6 +737,109 @@ function generatePolicyID() {
     return _.times(16, () => Math.floor(Math.random() * 16).toString(16)).join('').toUpperCase();
 }
 
+/**
+ * Optimistically creates a new workspace and default workspace chats
+ */
+function createWorkspace() {
+    const policyID = generatePolicyID();
+    const {
+        announceChatReportID,
+        announceChatData,
+        adminsChatReportID,
+        adminChatData,
+        expenseChatReportID,
+        expenseChatData,
+    } = Report.createOptimisticWorkspaceChats(policyID, sessionEmail);
+    const workspaceName = generateDefaultWorkspaceName();
+    API.write('CreateWorkspace', {
+        policyID,
+        announceChatReportID,
+        adminsChatReportID,
+        expenseChatReportID,
+        policyName: workspaceName,
+        type: CONST.POLICY.TYPE.FREE,
+    }, {
+        optimisticData: [{
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                id: policyID,
+                type: CONST.POLICY.TYPE.FREE,
+                name: workspaceName,
+                role: CONST.POLICY.ROLE.ADMIN,
+                owner: sessionEmail,
+                outputCurrency: 'USD',
+                pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+            },
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY_MEMBER_LIST}${policyID}`,
+            value: getSimplifiedEmployeeList([{email: sessionEmail, role: CONST.POLICY.ROLE.ADMIN}]),
```

Comment:
Theoretically we should not need this method in App going forward as we used to to massage data coming from server and that should already be coming in the correct shape.

Should we just already se this object to the correct format and avoid using this data format manipulating method?

Reply:
Do you have an issue link / PR for this? 

I am thinking we still need it locally if they want to update their workspace members offline, but then once we get online the changes should sync with the server and update the list

Reply:
I dont have an issue for this but I am just coming from the original refactoring ides to do no data manipulation on the frontend. That said, it concerns data getting from backend so in this case it is fine, but I am just wondering if we could already format this into the shape we need (given it is just one email and role it should be fine) to simplify stuff.

I am not suggesting to deprecate the method and delete it from the codebase, just not use it here :D

Reply:
Ah gotcha, yeah I can do that 👍

Reply:
Nice thanks!